### PR TITLE
[gdk_pixbuf] Update to 2.42.8, enable introspection for x86_64-linux-gnu

### PR DESCRIPTION
**Pull request by jwahlstrand**
Bump gdkpixbuf to 2.42.8 and include gobject-introspection data for x86_64-linux-gnu.

On Apple (both ARM and x86_64) I ran into a linker bug and copied a workaround from GTK4's build_tarballs.jl. (#3915)

My real motive here is to get more introspection data to use in Gtk4.jl. If this looks OK, I'd like to repeat this for libgtk's other GLib-based dependencies.

### CODE REVIEW

File: G/gdk_pixbuf/build_tarballs.jl [hunk 1]
```diff
@@ -49,19 +66,23 @@ products = [
...
+introspect_platform = filter(p -> triplet(p) == "x86_64-linux-gnu", platforms)
+
 # Dependencies that must be installed before this package can be built
 dependencies = [
     # Need a host gettext for msgfmt
     HostBuildDependency("Gettext_jll"),
     # Need a host glib for glib-compile-resources
     HostBuildDependency("Glib_jll"),
-    Dependency("Glib_jll"; compat="2.68.1"),
+    Dependency("Glib_jll"; compat="2.68.3"),
```
Comment (giordano) [inline G/gdk_pixbuf/build_tarballs.jl:78]: Is this change necessary?

File: G/gdk_pixbuf/build_tarballs.jl [hunk 2]
```diff
@@ -49,19 +66,23 @@ products = [
 # Some dependencies are needed only on Linux and FreeBSD
 linux_freebsd = filter(p->Sys.islinux(p)||Sys.isfreebsd(p), platforms)
 
+# gobject_introspection is needed only on x86_64-linux-gnu
+introspect_platform = filter(p -> triplet(p) == "x86_64-linux-gnu", platforms)
```
Comment (giordano) [inline G/gdk_pixbuf/build_tarballs.jl:70]: I prefer more declarative checks, checking the triplet is problematic in the case in which we need to expand the list of platforms above with more tags
```suggestion
introspect_platform = filter(p -> Sys.islinux(p) && libc(p) == "glibc" && arch(p) == "x86_64", platforms)
```

File: G/gdk_pixbuf/build_tarballs.jl (hunk 1 above)
Comment (jwahlstrand) [reply G/gdk_pixbuf/build_tarballs.jl:78]: That GLib version was set in the build for gobject_introspection (by me, oops) and I'm seeing:
```
ERROR: LoadError: Unsatisfiable requirements detected for package Glib_jll [7746bdde]:
 Glib_jll [7746bdde] log:
 ├─possible versions are: 2.59.0-2.68.3 or uninstalled
 ├─restricted to versions 2.68.1 by an explicit requirement, leaving only versions 2.68.1
 └─restricted by compatibility requirements with gobject_introspection_jll [6306888c] to versions: 2.68.3 — no versions left
   └─gobject_introspection_jll [6306888c] log:
     ├─possible versions are: 1.72.0 or uninstalled
     └─restricted to versions * by an explicit requirement, leaving only versions 1.72.0
```
There's no reason to require 2.68.3 in gobject_introspection_jll -- could we create a new version of that with 2.68.1?

Comment (giordano) [reply G/gdk_pixbuf/build_tarballs.jl:78]: Uhm, I see.  No, ok, that should be fine, I just wanted to understand what's going on.
